### PR TITLE
docs: add hydra cloud gsg

### PR DESCRIPTION
**Pull request by tomekpapiernik**
This PR adds a getting started guide for using the Cloud OAuth2 service (Hydra).

https://github.com/ory/docs/pull/981

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 1]
```diff
@@ -0,0 +1,213 @@
+---
+id: ory-cloud-oauth2
+title: Perform OAuth2 Authorization Code Grant with Ory Cloud
+sidebar_label: OAuth2 in Ory Cloud
+---
+
+This guide shows you a sample implementation of an Authorization Code Grant OAuth2 flow in Ory Cloud that you can use as a
```
Comment (kmherrmann) [inline docs/getting-started/ory-cloud-oauth2.mdx:7]: Can we add some context on

* What OAuth2 is for (3rd party API access, signing into 3rd party apps using your Ory identities ["becoming a social login provider"]
* The fact that Ory Cloud supports this out-of-the box incl. OIDC, Authorization Code Flows, Client Credentials Grant etc.
* A pointer to the in-depth-guides to the concepts in hydra docs (https://www.ory.sh/docs/hydra/concepts/before-oauth2)
* This guide will lead the user end-to-end through a first set-up, with the example of an Authorization Code Grant flow


Thanks!

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+For example, a service for selling goods allows users to add pictures from their Google Photos albums to their listings. To get
+access to the user's photos, the OAuth2 client of the goods selling service asks the user for consent to access data on their
+behalf. The request includes scopes - a list of operations the client wants to perform on behalf of the user. When the user
+agrees, the OAuth2 client gets an authorization code which it exchanges for an `id_token` and `access_token` pair issued by Google
+Photos. These tokens allow the client to access the user's resources from an external service.
+
+By following this guide you will:
+
+- Run a simple login and consent Node.js application on your machine.
```
Comment (kmherrmann) [inline docs/getting-started/ory-cloud-oauth2.mdx:21]: explain what it does: The Web UI to let users grant or reject access to their data

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+- Create an OAuth2 client in Ory Cloud.
```
Comment (kmherrmann) [inline docs/getting-started/ory-cloud-oauth2.mdx:22]: I would make this the 1st step

Comment (kmherrmann) [reply docs/getting-started/ory-cloud-oauth2.mdx:22]: (which it is in the flow below anyway)

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+2. On the login screen, provide the user details and log in to proceed.
+3. Choose the scopes you want to give the client access to and click **Allow access**. You must choose at least one scope.
+4. The next page shows the tokens the client got as a result of running the Authorization Code Grant flow.
+
+:::note
+
+You can get a closer look at the `access_token` by decoding it at [jwt.io](https://jwt.io/). In the decoded token, you can find
+information about the user on whose behalf the client acts and the URL of the issuer - your Ory Cloud project.
+
+:::
```
Comment (kmherrmann) [inline docs/getting-started/ory-cloud-oauth2.mdx:213]: Let's add a nice "that's it - you've now succeeded and your client now has access to ..." message to make sure the reader understood what just happend.

**Review by kmherrmann — CHANGES_REQUESTED** [review]
Thanks - this already looks neat! 
I've added a few points where I believe additional context is helpful to newcomers.

File: docs/getting-started/ory-cloud-oauth2.mdx (hunk 3 above)
Comment (tomekpapiernik) [reply docs/getting-started/ory-cloud-oauth2.mdx:22]: Good point, done.

**Comment by tomekpapiernik** [comment]
@kmherrmann I updated the document, please have another look when you get the chance. Thanks!

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 5]
```diff
@@ -0,0 +1,225 @@
+---
+id: ory-cloud-oauth2
+title: Perform OAuth2 Authorization Code Grant with Ory Cloud
+sidebar_label: OAuth2 in Ory Cloud
+---
+
+This guide shows you a sample implementation of an Authorization Code Grant OAuth2 flow in Ory Cloud that you can use as a
+starting point for using the Ory Cloud OAuth2 Service (Ory Hydra). It takes you on a complete journey, from initial setup to
```
Comment (kmherrmann) [inline docs/getting-started/ory-cloud-oauth2.mdx:8]: In the spirit of cloud-first and concise names, let's call the thing "Ory OAuth2 Federation" for now, and shorten to something like:

This guide shows you a sample implementation of an _Authorization Code Grant_ OAuth2 flow using Ory OAuth2 Federation* in Ory Cloud.

... then I'd like to relegate Ory Hydra to a footnote, as in: 
* Ory OAuth2 Federation is fully managed in Ory Cloud and based on the open source Ory Hydra federation server.

**Review by kmherrmann — COMMENTED** [review]
Great job, thank you so much!
Just one nit on the product naming and reference to Hydra, then good to merge!

File: docs/getting-started/ory-cloud-oauth2.mdx (hunk 5 above)
Comment (tomekpapiernik) [reply docs/getting-started/ory-cloud-oauth2.mdx]: Done, please see if the way I did this is okay for you :)

**Comment by tomekpapiernik** [comment]
The build step fails on a hydra cli file I haven't touched :(

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+  </TabItem>
+  <TabItem value="hydracli" label="Ory Hydra CLI">
+
+```shell
+hydra clients create \
+    --name YOUR_CLIENT_NAME \
+    --grant-types authorization_code,refresh_token \ # Mandatory for this example, don't change.
+    --response-types code,id_token \ # Mandatory for this example, don't change.
+    --access-token ORY_PAT \ # Ory PAT copied from the Cloud Console, 'ory_pat' prefix included.
+    --endpoint https://SDK_CONFIGURATION_URL/admin # Found in the 'Connect' section of the Cloud Console. Must end with '/admin'.
```
Comment (piotrmsc) [inline docs/getting-started/ory-cloud-oauth2.mdx:100]: we miss redirects uri as callback 
```
    --callbacks http://127.0.0.1:5555/callback
 ```

Comment (piotrmsc) [reply docs/getting-started/ory-cloud-oauth2.mdx:100]: and same for scopes
```
    --scope openid,offline \

```

Comment (tomekpapiernik) [reply docs/getting-started/ory-cloud-oauth2.mdx:101]: Fixed. I must've removed these by accident at some point, sorry.

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    ],
+    "redirect_uris": [
+      "http://127.0.0.1:5555/callback"
+    ],
+    "response_types": [
+          "code",
+          "id_token",
+          "refresh_token"
+    ],
+    "scope": "openid offline"
```
[the remark this answers is not in the document]
Comment (kmherrmann) [reply docs/getting-started/ory-cloud-oauth2.mdx:87]: Redirect URI (--callbacks is also missing)

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 8]
```diff
@@ -0,0 +1,228 @@
+---
+id: ory-cloud-oauth2
+title: Perform OAuth2 Authorization Code Grant with Ory Cloud
+sidebar_label: OAuth2 in Ory Cloud
```
Comment (kmherrmann) [inline docs/getting-started/ory-cloud-oauth2.mdx:4]: "Federation with OAuth2"

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+
+```shell
+hydra clients create \
+    --name YOUR_CLIENT_NAME \
+    --grant-types authorization_code,refresh_token \ # Mandatory for this example, don't change.
+    --response-types code,id_token \ # Mandatory for this example, don't change.
+    --scope openid,offline \ # Mandatory for this example, don't change.
+    --access-token ORY_PAT \ # Ory PAT copied from the Cloud Console, 'ory_pat' prefix included.
+    --endpoint https://SDK_CONFIGURATION_URL/admin # Found in the 'Connect' section of the Cloud Console. Must end with '/admin'.
+    --callbacks http://127.0.0.1:5555/callback \ # Mandatory for this example, don't change.
```
Comment (aeneasr) [inline docs/getting-started/ory-cloud-oauth2.mdx:102]: Trailing `\` means that there is another line coming :) If it is the last line, omit it:

```suggestion
    --callbacks http://127.0.0.1:5555/callback # Mandatory for this example, don't change.
```

File: docs/hydra/cli/hydra-janitor.md [hunk 10]
```diff
@@ -31,7 +31,7 @@ this command with caution.
 Janitor can be used in several ways.
 
 1. By passing the database connection string (DSN) as an argument
-   Pass the database url (dsn) as an argument to janitor. E.g. janitor <database-url>
+   Pass the database url (dsn) as an argument to janitor. E.g. janitor {database-url}
```
Comment (aeneasr) [inline docs/hydra/cli/hydra-janitor.md:34]: This you will also need to fix in Ory Hydra itself :) Otherwise it will be overwritten on next push ...

**Review by aeneasr — COMMENTED** [review]
Great stuff! I have not tried it out yet but it looks very good. Just two code-related things from my side!

File: docs/hydra/cli/hydra-janitor.md (hunk 10 above)
Comment (tomekpapiernik) [reply docs/hydra/cli/hydra-janitor.md:34]: Right! I'll try to locate this file and create PR with fix in the CLI code. Thanks

File: docs/getting-started/ory-cloud-oauth2.mdx (hunk 9 above)
Comment (tomekpapiernik) [reply docs/getting-started/ory-cloud-oauth2.mdx:102]: Good point, I missed that. Thanks!

**Pull request merged**